gazelle_haskell_modules/rule_generation.go

```diff
@@ -393,6 +418,26 @@ func getSources(expr build.Expr) (map[string]bool, error) {
 	return sourceMap, nil
 }
 
+func getSourcesRecursivelyFromDirs(pkgRoot string, dirs []string) ([]string, error) {
+	var srcs []string
+	for _, dir := range dirs {
+		err := filepath.WalkDir(path.Join(pkgRoot, dir), func(path string, d fs.DirEntry, err error) error {
```

Comment:
Should I use gazelle's `walk` instead here? I don't think I need anything from it, but it does some special things to directories containing BUILD files, afaiu.

Reply:
I guess it would be a good idea since it allows user to add some configuration specific to subdirectories. I don't know how that config side works though, and how it will affect Haskell specific config.